oscillators.lib

```diff
@@ -446,6 +446,52 @@ lf_sawpos_phase(phase,freq) = (+(phase-phase') : ma.frac) ~ +(freq/ma.SR);
 // MarkDown: Romain Michon
 lf_sawpos(freq) = ma.frac ~ +(freq/ma.SR);
 
+//-----------------`(os.)lf_sawpos_reset`--------------------
+// Simple sawtooth waveform oscillator between 0 and 1.
+// with reset.
+//
+// #### Usage
+//
+// ```
+// lf_sawpos_reset(freq,reset)
+// ```
+//
+// Where:
+//
+// * `freq`: frequency
+// * `reset`: reset the oscillator to 0
+//
+//---------------------------------------------------------
+// Author: Bart Brouns
+// License: STK-4.3
+lf_sawpos_reset(freq,reset) = ma.frac * dontReset ~ +(freq/ma.SR)
+with {
+  dontReset  = reset:ba.impulsify*-1+1;
```

Comment:
Why is this dontReset so complicated? Can you explain what signal it produces?

Reply:
It's a stream on 1's,with a single sample zero when reset goes from 0 to bigger then 0.
Can you think of a way to simplify it?